### PR TITLE
Added recursive download and tab icon when a filesystem is shared

### PR DESCRIPTION
**Pull request by Pourliver**
We can now right click and download a folder recursively. I made the downloadDialog non-modal, since a modal is a blocking action, meaning that we cannot focus the original window. I did the same thing to the download completion popup.

I also added a jobs queue that allows us to queue up downloads, they starts as soon as the previous download has ended. Practical when we are downloading a huge zip / folder, and we want to keep looking for interesting file.

The logic may be confusing, because of the nature of the observers, I had to make sure that only one job was executing at a time.

The 3 queues have distinct jobs. `jobsQueue` holds the action requested by the user. When a job is being processed, the sub-jobs goes into `directoryDownloadQueue` and `fileDownloadQueue`. When we download a folder, we start by downloading all files inside the folder, then we do the same for each sub-folder.

I've worked on this for a while so I don't know what I should explain design wise lol. Let me know if anything is unclear.

### CODE REVIEW

File: pyrdp/player/LiveEventHandler.py [hunk 1]
```diff
@@ -80,6 +90,46 @@ def onListDirectory(self, deviceID: int, path: str):
...
+    def addToDownloadQueue(self, item: Union[File, Directory], targetPath: str, dialog: FileDownloadDialog):
+        job = (item, targetPath, dialog)
+
+        self.jobsQueue.add(job)
+
+        if self.currentDownload == None:
+            self.dispatchDownload()
+
+    def dispatchDownload(self):
+        # We download each file of a directory before enumerating other directories.
```
Comment (Res260) [inline pyrdp/player/LiveEventHandler.py]: docstring ` ``` ` instead of comment `#`

File: pyrdp/player/LiveEventHandler.py [hunk 2]
```diff
@@ -80,6 +90,46 @@ def onListDirectory(self, deviceID: int, path: str):
...
+        # Otherwise we may fill the stack with too many queued items.
+        # When we're done with both file and directories, we start an other queued job
+
+        # Request download of a queued file
+        if len(self.fileDownloadQueue) != 0:
```
Comment (Res260) [inline pyrdp/player/LiveEventHandler.py]: `myQueue.empty()` exists I think

Comment (Res260) [reply pyrdp/player/LiveEventHandler.py]: (see elifs as well)

File: pyrdp/player/LiveEventHandler.py [hunk 3]
```diff
@@ -139,7 +217,26 @@ def onFileDownloadRequested(self, file: File, targetPath: str, dialog: FileDownl
         self.downloadFiles[remotePath] = targetFile
         self.downloadDialogs[remotePath] = dialog
 
-        pdu = PlayerFileDownloadRequestPDU(self.layer.getCurrentTimeStamp(), parent.deviceID, file.getFullPath())
+        pdu = PlayerFileDownloadRequestPDU(self.layer.getCurrentTimeStamp(), parent.deviceID, remotePath)
+        self.layer.sendPDU(pdu)
+
+    def onDirectoryDownloadRequested(self, directory: Directory, targetPath: str, dialog: FileDownloadDialog):
```
Comment (Res260) [inline pyrdp/player/LiveEventHandler.py]: since this method got way bigger docstring would help

File: pyrdp/player/LiveEventHandler.py (hunk 2 above)
Comment (Pourliver) [reply pyrdp/player/LiveEventHandler.py]: It doesn't, in python `if not myQueue` is falsy if the queue is empty. Even though its very pythonic, it's not very verbose. I think using `len(myQueue) != 0` should stay, as seen everywhere else in the code base

File: pyrdp/player/LiveEventHandler.py (hunk 3 above)
Comment (Pourliver) [reply pyrdp/player/LiveEventHandler.py]: Will do! I'll add more documentation to the flow of the module in general.

File: pyrdp/player/LiveEventHandler.py (hunk 2 above)
Comment (Res260) [reply pyrdp/player/LiveEventHandler.py]: im confused: https://docs.python.org/3/library/queue.html#queue.Queue.empty

Comment (Pourliver) [reply pyrdp/player/LiveEventHandler.py]: Thats the definition of the queue class, which is not what I'm using. The Queue class is meant for multi-threaded application.

Comment (Res260) [reply pyrdp/player/LiveEventHandler.py]: Damn, I overlooked that! I understand.

**Comment by Res260** [comment]
Merged master into the branch, updated changelog and tested. Ready to merge.

**Pull request merged**